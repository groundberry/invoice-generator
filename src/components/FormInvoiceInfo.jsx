import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const styles = theme => ({
    textField: {
      display: "flex"
    },
  });

class FormInvoiceInfo extends React.Component {
  handleChange = evt => {
    const { name, value } = evt.target;
    this.props.onChange({
      [name]: value
    });
  };

  render() {
    const {
        classes,
        number,
        emitDate,
        terms,
        dueDate
    } = this.props;

    return (
      <form noValidate autoComplete="off">
        <TextField
          label="Invoice Number"
          className={classes.textField}
          name="number"
          value={number}
          onChange={this.handleChange}
          type="number"
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Date"
          className={classes.textField}
          name="emitDate"
          value={emitDate}
          onChange={this.handleChange}
          type="date"
          InputLabelProps={{
            shrink: true
          }}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Terms"
          className={classes.textField}
          name="terms"
          value={terms}
          onChange={this.handleChange}
          margin="normal"
          variant="outlined"
        />
        <TextField
          label="Due"
          className={classes.textField}
          name="dueDate"
          value={dueDate}
          onChange={this.handleChange}
          type="date"
          InputLabelProps={{
            shrink: true
          }}
          margin="normal"
          variant="outlined"
        />
      </form>
    );
  }
}

FormInvoiceInfo.propTypes = {
  classes: PropTypes.object.isRequired,
  number: PropTypes.string.isRequired,
  emitDate: PropTypes.string.isRequired,
  terms: PropTypes.string.isRequired,
  dueDate: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default withStyles(styles)(FormInvoiceInfo);
