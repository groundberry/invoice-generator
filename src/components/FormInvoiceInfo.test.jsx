import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import FormInvoiceInfo from "./FormInvoiceInfo";
import TextField from "@material-ui/core/TextField";

const data = {
  number: "0001",
  emitDate: "04/10/2019",
  terms: "30 Days",
  dueDate: "05/10/2019"
};

describe("FormInvoiceInfo", () => {
  let wrapper;
  let onChange;

  beforeEach(() => {
    onChange = jest.fn();

    const shallow = createShallow({
      untilSelector: FormInvoiceInfo
    });
    wrapper = shallow(
      <FormInvoiceInfo
        number={data.number}
        emitDate={data.emitDate}
        terms={data.terms}
        dueDate={data.dueDate}
        onChange={onChange}
      />
    );
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when invoice number changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "number" })
        .simulate("change", {
          target: { name: "number", value: "0002" }
        });
    });

    it("invokes onChange with the new number", () => {
      expect(onChange).toHaveBeenCalledWith({ number: "0002" });
    });
  });

  describe("when emit date changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "emitDate" })
        .simulate("change", {
          target: { name: "emitDate", value: "04/11/2019" }
        });
    });

    it("invokes onChange with the new emit date", () => {
      expect(onChange).toHaveBeenCalledWith({ emitDate: "04/11/2019" });
    });
  });

  describe("when terms change", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "terms" })
        .simulate("change", { target: { name: "terms", value: "31 Days" } });
    });

    it("invokes onChange with the new terms", () => {
      expect(onChange).toHaveBeenCalledWith({ terms: "31 Days" });
    });
  });

  describe("when due date changes", () => {
    beforeEach(() => {
      wrapper
        .find(TextField)
        .filter({ name: "dueDate" })
        .simulate("change", { target: { name: "dueDate", value: "05/12/2019" } });
    });

    it("invokes onChange with the new due date", () => {
      expect(onChange).toHaveBeenCalledWith({ dueDate: "05/12/2019" });
    });
  });
});
