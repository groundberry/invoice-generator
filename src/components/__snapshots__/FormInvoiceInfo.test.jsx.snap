// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInvoiceInfo renders 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <TextField
    className="FormInvoiceInfo-textField-1"
    label="Invoice Number"
    margin="normal"
    name="number"
    onChange={[Function]}
    required={false}
    select={false}
    type="number"
    value="0001"
    variant="outlined"
  />
  <TextField
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    className="FormInvoiceInfo-textField-1"
    label="Date"
    margin="normal"
    name="emitDate"
    onChange={[Function]}
    required={false}
    select={false}
    type="date"
    value="04/10/2019"
    variant="outlined"
  />
  <TextField
    className="FormInvoiceInfo-textField-1"
    label="Terms"
    margin="normal"
    name="terms"
    onChange={[Function]}
    required={false}
    select={false}
    value="30 Days"
    variant="outlined"
  />
  <TextField
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    className="FormInvoiceInfo-textField-1"
    label="Due"
    margin="normal"
    name="dueDate"
    onChange={[Function]}
    required={false}
    select={false}
    type="date"
    value="05/10/2019"
    variant="outlined"
  />
</form>
`;
